pkg/reduce/pnf/processandforward.go

```diff
@@ -213,9 +219,16 @@ func (p *ProcessAndForward) writeToBuffer(ctx context.Context, bufferID string,
 		offsets, writeErrs = p.toBuffers[bufferID].Write(ctx, writeMessages)
 		for i, message := range writeMessages {
 			if writeErrs[i] != nil {
-				failedMessages = append(failedMessages, message)
+				if errors.As(writeErrs[i], &isb.NoRetryableBufferWriteErr{}) {
+					// If toBuffer returns us a NoRetryableBufferWriteErr, we drop the message.
+					dropBytes += float64(len(message.Payload))
+					p.log.Debugw("Dropped message", zap.String("bufferID", bufferID), zap.Any("message", message), zap.Error(writeErrs[i]))
```

Comment:
Do we do the same logging in map? One concern is this will generate lots of logs for a batch.

Reply:
No we don't. I think I added this line because in reduce forwarder, when writing successfully, we have such log enabled.

```
p.log.Debugw("Forwarded message", zap.String("bufferID", bufferID), zap.Any("message", message))
```

Do we know why we have it for reduce but not map? I also noticed the log is at debug level so maybe it's configured this way intentionally?